src/main/java/core/basesyntax/strategy/ReturnOperationHandler.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.serviceimpl.FruitTransactionValidatorImpl;
+import java.util.Optional;
+
+public class ReturnOperationHandler implements OperationHandler {
+    private FruitTransactionValidatorImpl validator;
+
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        validator.validateOperation(fruitTransaction);
+        String fruit = fruitTransaction.getFruit();
+        validator.validateFruit(fruit);
+        int amountToReturn = fruitTransaction.getQuantity();
+        validator.validateAmount(amountToReturn, FruitTransaction.Operation.RETURN);
```

Comment:
It is redundant to validate the amount with the 'RETURN' operation type since this handler should only deal with the 'RETURN' operation. The validation should be generic or already handled before calling the apply method.